scorecard/profiles.py

```diff
@@ -168,40 +124,25 @@ def get_profile(geo_code, geo_level, profile_name=None):
         results[item] = requests.get(url, verify=False).json()
 
 
-    op_exp_actual = get_quarter_results(results['op_exp_actual'])
-    op_exp_budget = results['op_exp_budget']['cells'][0]['amount.sum']
-    op_rev = get_quarter_results(results['op_rev'])
-
-    cap_exp_actual = get_quarter_results(
-        results['cap_exp_actual'],
-        line_items['cap_exp_actual']['aggregate'])
-    cap_exp_budget = results['cap_exp_budget']['cells'][0][line_items['cap_exp_budget']['aggregate']]
-    cap_rev = get_quarter_results(results['cap_rev'], line_items['cap_rev']['aggregate'])
-
-    cash_flow = [
-        r['amount.sum'] for r in results['cash_flow']['cells']
-        if r['financial_period.period'] == current_month][0]
+    op_exp_actual = amount_from_results('op_exp_actual', results, line_items)
+    op_exp_budget = amount_from_results('op_exp_budget', results, line_items)
 
-    debtors = [
-        r['total_amount.sum'] for r in results['debtors']['cells']
-        if r['financial_period.period'] == current_month][0]
+    cap_exp_actual = amount_from_results('cap_exp_actual', results, line_items)
+    cap_exp_budget = amount_from_results('cap_exp_budget', results, line_items)
 
-    cap_grant = get_quarter_results(results['cap_grant'])
-    op_grant = get_quarter_results(results['op_grant'])
+    cash_flow = amount_from_results('cash_flow', results, line_items)
 
-    rep_maint = get_quarter_results(results['rep_maint']) or 0
-    ppe = results['ppe']['cells'][0][line_items['ppe']['aggregate']] or 0
-    invest_prop = results['invest_prop']['cells'][0][line_items['ppe']['aggregate']] or 0
+    rep_maint = amount_from_results('rep_maint', results, line_items)
+    ppe = amount_from_results('ppe', results, line_items)
+    invest_prop = amount_from_results('invest_prop', results, line_items)
 
-    debtors_perc_rev = debtors / ((cap_rev+op_rev) - (cap_grant + op_grant)) * 100
-    cash_coverage = cash_flow / (op_exp_actual / 12)
-    op_budget_diff = (op_exp_actual - op_exp_budget) / op_exp_budget if op_exp_budget else 0
-    cap_budget_diff = (cap_exp_actual - cap_exp_budget) / cap_exp_budget if cap_exp_budget else 0
-    rep_maint_perc_ppe = rep_maint * 12 / (ppe + invest_prop)
+    cash_coverage = cash_flow / (op_exp_actual / 12) if op_exp_actual > 0 else 'N/A'
```

Comment:
we can use wazimap.data.utils.percent and wazimap.data.utils.ratio if we need to, that handles zero denominators.